codegen/src/main/java/software/amazon/awssdk/codegen/naming/DefaultNamingStrategy.java

```diff
@@ -70,34 +77,102 @@ private static boolean isJavaKeyword(String word) {
     }
 
     @Override
-    public String getExceptionName(String errorShapeName) {
+    @ReviewBeforeRelease("Always use service ID as the base.")
+    public String getServiceName() {
+        String baseName = Stream.of(serviceModel.getMetadata().getServiceId(),
```

Comment:
What's the plan for removing the ReviewBeforeRelease annotation? If we are going to throw an error eventually if not present, sounds good. If we are ever going to have a fallback lets log a warning.

Reply:
We'll be throwing an exception in the future (before GA), but right now the service ID isn't present for all of the services.